library/common.h

```diff
@@ -59,27 +59,64 @@
 
 /** Byte Reading Macros
  *
- * To tidy up code and save horizontal and vertical space, use byte
- * reading macros to cast
+ * Obtain the most significant byte of x using 0xff
+ * Using MBEDTLS_BYTE_a will shift a*8 bits
+ * to retrieve the next byte of information
  */
 #define MBEDTLS_BYTE_0( x ) ( (uint8_t) ( ( x ) & 0xff ) )
 #define MBEDTLS_BYTE_1( x ) ( (uint8_t) ( ( ( x ) >> 8  ) & 0xff ) )
 #define MBEDTLS_BYTE_2( x ) ( (uint8_t) ( ( ( x ) >> 16 ) & 0xff ) )
 #define MBEDTLS_BYTE_3( x ) ( (uint8_t) ( ( ( x ) >> 24 ) & 0xff ) )
 
-/*
- * 32-bit integer manipulation macros (big endian)
+#define MBEDTLS_CHAR_0( x ) ( (unsigned char) (   ( x )         & 0xff ) )
```

Comment:
I just replaced a few instances of `MBEDTLS_CHAR` with `MBEDTLS_BYTE` in a few different files but I get some errors (below) which I find odd as `common.h` contains `#include "mbedtls/config.h"`
When I was choosing BYTE or CHAR for each line, I was looking at whether it was originally casted or what type the data was being stored in. I figured it would be a bad idea to implicitly cast the data, but if they are equivalent I can try work around this error
```  
CC    ccm.c
In file included from ccm.c:29:
ccm.c: In function ‘ccm_auth_crypt’:
common.h:66:32: error: ‘uint8_t’ undeclared (first use in this function); did you mean ‘u_int8_t’?
   66 | #define MBEDTLS_BYTE_0( x ) ( (uint8_t) ( ( x ) & 0xff ) )
      |                                ^~~~~~~
ccm.c:203:19: note: in expansion of macro ‘MBEDTLS_BYTE_0’
  203 |         b[15-i] = MBEDTLS_BYTE_0( len_left );
      |                   ^~~~~~~~~~~~~~
common.h:66:32: note: each undeclared identifier is reported only once for each function it appears in
   66 | #define MBEDTLS_BYTE_0( x ) ( (uint8_t) ( ( x ) & 0xff ) )
      |                                ^~~~~~~
ccm.c:203:19: note: in expansion of macro ‘MBEDTLS_BYTE_0’
  203 |         b[15-i] = MBEDTLS_BYTE_0( len_left );
      |                   ^~~~~~~~~~~~~~
make[1]: *** [Makefile:269: ccm.o] Error 1
make: *** [Makefile:18: lib] Error 2```

Reply:
I think you just have to include stdint.h (it defined uint8_t) in common.h. You have to do it if you use the `uint8_t` in common.h.

Reply:
That makes sense, should be sorted. I will do the same for the 2.x backport as with the other suggestions below